Auth0/Authentication.swift

```diff
@@ -1080,6 +1080,57 @@ public protocol Authentication: Trackable, Loggable {
      */
     func jwks() -> Request<JWKS, AuthenticationError>
 
+    /**
+     Performs a custom token exchange to obtain Auth0 credentials using an existing identity provider token.
+
+     ## Usage
+
+     ```swift
+     Auth0
+         .authentication()
+         .customTokenExchange(subjectToken: "existing-token",
+                              subjectTokenType: "urn:ietf:params:oauth:token-type:jwt",
+                              audience: "https://example.com/api",
+                              scope: "openid profile email")
+         .start { result in
+             switch result {
+             case .success(let credentials):
+                 print("Obtained credentials: \(credentials)")
+             case .failure(let error):
+                 print("Failed with: \(error)")
+             }
+         }
+     ```
+
+     You can also include additional parameters:
+
+     ```swift
+     Auth0
+         .authentication()
+         .customTokenExchange(subjectToken: "existing-token",
+                              subjectTokenType: "urn:ietf:params:oauth:token-type:jwt",
+                              audience: "https://example.com/api",
+                              scope: "openid profile email",
+                              additionalParameters: ["custom_claim": "value"])
+         .start { print($0) }
+     ```
+
+     - Parameters:
+       - subjectToken: The security token to be exchanged.
+       - subjectTokenType: URI that identifies the type of the subject token.
+       - audience: API Identifier that your application is requesting access to. Defaults to `nil`.
+       - scope: Space-separated list of requested scope values. Defaults to `openid profile email`.
+       - additionalParameters: Additional parameters to include in the token exchange request. Defaults to empty dictionary.
+     - Returns: A request that will yield Auth0 user's credentials.
+
+     ## See Also
+
+     - [Authentication API Endpoint](https://auth0.com/docs/api/authentication/token-exchange)
+     - [Custom Token Exchange Documentation](https://auth0.com/docs/authenticate/custom-token-exchange)
+     - [RFC 8693: OAuth 2.0 Token Exchange](https://tools.ietf.org/html/rfc8693)
+     */
+    func customTokenExchange(subjectToken: String, subjectTokenType: String, audience: String?, scope: String, additionalParameters: [String: Any]) -> Request<Credentials, AuthenticationError>
```

Comment:
Custom parameters (`additionalParameters`) are already supported. The `Request` type has a `parameters()` method for this purpose: https://github.com/auth0/Auth0.swift/blob/master/Auth0/Request.swift#L97

Reply:
Since the method is in the `Request` type, it's available for all API client methods that return a `Request`.

Reply:
Done